ebi_eva_common_pyutils/nextflow/nextflow_pipeline.py

```diff
@@ -0,0 +1,181 @@
+        if process_dependency_map:
+            self.add_dependencies(process_dependency_map)
+
+    def add_dependencies(self, process_dependency_map: Dict[NextFlowProcess, List[NextFlowProcess]]):
+        for process, dependencies in process_dependency_map.items():
+            if dependencies:
+                for dependency in dependencies:
+                    self.add_process_dependency(process, dependency)
+            else:
+                self.add_process_dependency(process, None)
+
+    def add_process_dependency(self, process: NextFlowProcess, dependency: Union[NextFlowProcess, None]):
+        if dependency:
+            self.process_dependency_map.add_edge(process, dependency)
+            if not nx.dag.is_directed_acyclic_graph(self.process_dependency_map):
+                raise ValueError(f"Cycles found in pipeline when adding process {process.process_name} "
+                                 f"and its dependency {dependency.process_name}")
+        else:
+            # If no dependency is specified, the process will just be a single node in the DAG
+            self.process_dependency_map.add_node(process)
+
+    def _write_to_pipeline_file(self, workflow_file_path: str):
+        with open(workflow_file_path, "a") as pipeline_file_handle:
+            pipeline_file_handle.write(self.__str__() + "\n")
+
+    def run_pipeline(self, workflow_file_path: str, nextflow_binary_path: str = 'nextflow',
+                     nextflow_config_path: str = None, working_dir: str = ".", resume: bool = False,
+                     other_args: dict = None):
+        # Remove pipeline file if it already exists
+        if os.path.exists(workflow_file_path):
+            os.remove(workflow_file_path)
+        self._write_to_pipeline_file(workflow_file_path)
+        workflow_command = f"cd {working_dir} && {nextflow_binary_path} run {workflow_file_path}"
+        workflow_command += f" -c {nextflow_config_path}" if nextflow_config_path else ""
+        workflow_command += f" -with-report {workflow_file_path}.report.html"
+        workflow_command += f" -with-dag {workflow_file_path}.dag.png"
+        workflow_command += " -resume" if resume else ""
+        workflow_command += " ".join([f" -{arg} {val}" for arg, val in other_args.items()]) if other_args else ""
+        run_command_with_output(f"Running pipeline {workflow_file_path}...", workflow_command)
+
+    @staticmethod
+    def join_pipelines(main_pipeline: 'NextFlowPipeline', dependent_pipeline: 'NextFlowPipeline') -> 'NextFlowPipeline':
+        joined_pipeline = NextFlowPipeline()
+        # Aggregate dependency maps of both pipelines
+        joined_pipeline.process_dependency_map = nx.compose(main_pipeline.process_dependency_map,
+                                                            dependent_pipeline.process_dependency_map)
+        # Every root process in the dependent pipeline depends on the final processes in the main pipeline
```

Comment:
I would promote this comment to a docstring, since to me at least it's not totally clear what a joined pipeline would be - something like "Returns a new pipeline whose root processes are those of the main pipeline, final processes are those of the dependent pipeline, and every root process of the dependent pipeline depends on the final processes of the main pipeline".